docs/collection_metadata.rst

```diff
@@ -26,6 +26,29 @@ The short-hand form of the command is also supported::
 
     ansible-builder introspect --sanitize ~/.ansible/collections/
 
+The ``--sanitize`` option will go through all of the collection requirements and
+remove duplicates, as well as remove some Python requirements that should normally
+be excluded. For instance, these Python requirements will *always* be excluded
+since they should be a part of execution environment definition itself:
+
+   - ansible
```

Comment:
I guess this depends on the version of ansible.

After 2.9, the ansible package isn't strictly necessary and this is reflected in the published images for ansible-runner, for example:
```
# for version in 2.12 2.11 2.10 2.9
> do
> podman run -it quay.io/ansible/ansible-runner:stable-${version}-devel pip freeze | grep ansible
> done
ansible-core @ file:///output/wheels/ansible_core-2.12.1.post0-py3-none-any.whl
ansible-runner @ file:///output/wheels/ansible_runner-2.1.2.dev2-py3-none-any.whl
ansible-core @ file:///output/wheels/ansible_core-2.11.7.post0-py3-none-any.whl
ansible-runner @ file:///output/wheels/ansible_runner-2.1.2.dev2-py3-none-any.whl
ansible-base @ file:///output/wheels/ansible_base-2.10.16.post0-py3-none-any.whl
ansible-runner @ file:///output/wheels/ansible_runner-2.1.1.dev14-py3-none-any.whl
ansible @ file:///output/wheels/ansible-2.9.27.post0-py3-none-any.whl
ansible-runner @ file:///output/wheels/ansible_runner-2.1.2.dev2-py3-none-any.whl
```

After 2.10, the ``ansible`` package on pypi moved to a model where it only includes a set of collections.
My use case was to mirror the UX of installing 'ansible' from PyPI and then bumped into [that issue](https://github.com/ansible/ansible-builder/issues/323#issuecomment-1000003661) (thanks for pointing me here)

I don't mind too much because there's simple workarounds like mentioned in that issue in addition to installing collections "manually" with a galaxy [requirements.yml](https://github.com/ansible-community/ansible-build-data/blob/main/5/galaxy-requirements.yaml).

I could have missed it but it would be great to print a message that says a package was "sanitized" to avoid the effect of surprise.

Reply:
> I could have missed it but it would be great to print a message that says a package was "sanitized" to avoid the effect of surprise.

I think that was attempted as a comment in the resultant requirements file, but with subsequent refactoring with the assemble script and other stuff, I think that file may wind up getting trashed, as more things were moved inside of ephemeral build stages.

https://github.com/ansible/ansible-builder/blob/87ae1d6a57cd6fca6d483929959f67527432af58/ansible_builder/requirements.py#L47

Reply:
@AlanCoding would it make sense to bump the verbosity of that from debug to warning, perhaps ?